Fix iothub docs

myc2h6o opened the pull request:
To resolve comments from #15466 on existing iot hub document

stephybun commented on website/docs/r/iothub.html.markdown:
```diff
@@ -190,13 +190,13 @@ An `endpoint` block supports the following:
+* `identity_id` - (Optional) The ID of the User Managed Identity used to authenticate against the endpoint.
 
--> **NOTE:** `identity_id` can only be specified when `authentication_type` is `identityBased`. It must be one of the `identity_ids` of the Iot Hub. If not specified when `authentication_type` is `identityBased`, System Assigned Managed Identity of the Iot Hub will be used.
+-> **NOTE:** `identity_id` can only be specified when `authentication_type` is `identityBased`. It must be one of the `identity_ids` of the Iot Hub. If `identity_id`is omitted when `authentication_type` is `identityBased`, then the System Assigned Managed Identity of the Iot Hub will be used.
```
```suggestion
-> **NOTE:** `identity_id` can only be specified when `authentication_type` is `identityBased`. It must be one of the `identity_ids` of the IoT Hub. If `identity_id` is omitted when `authentication_type` is `identityBased`, then the System Assigned Managed Identity of the IoT Hub will be used.
```

FredrikAugust replied:
```suggestion
-> **NOTE:** `identity_id` can only be specified when `authentication_type` is `identityBased`. It must be one of the `identity_ids` of the Iot Hub. If `identity_id`is omitted when `authentication_type` is `identityBased`, then the system-assigned managed identity of the IoT Hub will be used.
```

Not a proper noun, and Microsoft write "system-assigned" with a hypen.

stephybun commented on website/docs/r/iothub.html.markdown:
```diff
@@ -190,13 +190,13 @@ An `endpoint` block supports the following:
+-> **NOTE:** `identity_id` can only be specified when `authentication_type` is `identityBased`. It must be one of the `identity_ids` of the Iot Hub. If `identity_id`is omitted when `authentication_type` is `identityBased`, then the System Assigned Managed Identity of the Iot Hub will be used.
 
-~> **NOTE:** System Assigned Managed Identity can only be used in an update because access to the endpoint cannot be granted before the creation is done. The extracted resources `azurerm_iothub_endpoint_*` can be used to create endpoints with System Assigned Managed Identity without the need of an update. 
+~> **NOTE:** An IoT Hub can only be updated to use the System Assigned Managed Identity for `endpoint` since it is not possible to grant access to the endpoint until after creation. The extracted resources `azurerm_iothub_endpoint_*` can be used to configure Endpoint with the IoT Hub's System Assigned Managed Identity without the need for an update.
```
```suggestion
~> **NOTE:** An IoT Hub can only be updated to use the System Assigned Managed Identity for `endpoint` since it is not possible to grant access to the endpoint until after creation. The extracted resources `azurerm_iothub_endpoint_*` can be used to configure Endpoints with the IoT Hub's System Assigned Managed Identity without the need for an update.
```

FredrikAugust replied:
```suggestion
~> **NOTE:** An IoT Hub can only be updated to use the system-assigned managed identity for `endpoint` since it is not possible to grant access to the endpoint until after creation. The extracted resources `azurerm_iothub_endpoint_*` can be used to configure endpoints with the IoT Hub's system-assigned managed identity without the need for an update.
```

stephybun reviewed (commented):
Appreciate you fixing up the docs @myc2h6o! Two minor comments on the notes but otherwise this should be good to go.

myc2h6o commented:
Hi @FredrikAugust @stephybun Thanks for reviewing this change! I've fixed the docs. Please take a look.
I also merged the main branch in order to apply the same fix in the newly merged change as well.
For `system-assigned managed identity`, docs in this provider seems to prefer capitalized case for these special words, so I kept the case and added the dash to make it `System-Assigned Managed Identity`

stephybun reviewed (approved):
Thanks @myc2h6o, LGTM 👍🏼
